### ISSUE TITLE
Testing going forward.

### ISSUE THREAD

**Issue by sergioisidoro** [milesmcc/shynet#167]
While navigating the project, and seeing what I could do to help, I noticed that the project has no tests.

Is this a conscious choice, or would you be open for some contributions to add tests?

**Comment by milesmcc** [comment]
No, I would absolutely love for the project to have more testing. It’s not a conscious choice at all — in fact, it’s the biggest thing I think the project needs right now.

It would be awesome to have a mix of integration and unit tests. I have a checklist of actions I manually test before each release (verify setting Shynet up from scratch works, create a new service, verify hits and sessions are being recorded, etc.) and it would be fantastic to automate that process.

**Comment by sumit4613** [comment]
@milesmcc Is anyone working on tests currently, if not I would love to work on those.

**Comment by milesmcc** [comment]
@sumit4613 go for it!
